.github/workflows/support-window.yml

```diff
@@ -0,0 +1,54 @@
+name: Update support window diagram
+on:
+  # When translations are added/removed
+  push:
+    paths: README*.md
+  # Manually, when TypeScript is released
+  # https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow
+  workflow_dispatch:
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          personal <- Sys.getenv("R_LIBS_USER")
+          dir.create(personal, recursive = TRUE)
+          install.packages(c("ggplot2", "svglite"), personal)
+        shell: Rscript {0}
+      - name: Fetch TypeScript versions and release dates from npm
+        run: |
+          npm view --json typescript time |
+            jq '
+              to_entries
+              # Earliest non-pre-release <major>.<minor>
+              | map(
+                (.key |= capture("^(?<version>[0-9]+\\.[0-9]+)\\.[0-9]+$").version)
+                | select(.key)
+              )
+              | unique_by(.key)
+              | from_entries
+            ' > docs/support-window.json
+      - run: git rm --ignore-unmatch 'docs/support-window*.svg'
+      - name: Make SVG diagram
+        run: |
+          export LOCPATH=.locale
+          mkdir "$LOCPATH"
+          for readme in README*.md; do (
+            if [ "$readme" != README.md ]; then
+              LANG=${readme:7:-3}
```

Comment:
Sidenote: using `${foo%...}` and `${foo#...}` or `=~` regexps is more robust than relying on char positions.

Reply:
Nice! I forgot about `=~` and [BASH_REMATCH](https://www.gnu.org/software/bash/manual/bash#index-BASH_005fREMATCH). That's definitely superior. :+1: